__tests__/modern.test.ts

```diff
@@ -449,3 +413,46 @@ it('Wrong key mentioned in theme configuration', () => {
     }
   );
 });
+
+it('With component Config', () => {
+  const config = {
+    default: {
+      light: {
+        background: 'purple',
+        extras: 'black'
+      },
+      dark: {
+        background: 'black'
+      }
+    },
+    mint: {
+      background: 'teal'
+    }
+  };
+
+  return run(
+    `
+      .test {
+        color: @theme background;
+        background-image: linear-gradient(to right, @theme background, @theme background)
+      }
+    `,
+    `
+      .test {
+        color: var(--background, yellow);
+        background-image: linear-gradient(to right, var(--background, yellow), var(--background, yellow))
+      }
+
+      .dark {
+        --background: pink
+      }
+      .mint.light {
+        --background: teal
+      }
+    `,
+    {
+      config
+    },
+    './__tests__/test-modern-themes-ts/test.css'
+  );
+});
```

Comment:
We have no test cases that actually test with a component config

Reply:
🎉